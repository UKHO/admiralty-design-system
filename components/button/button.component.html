<button [disabled]="disabled" class="{{ secondary ? 'secondary' : '' }}">
  <ng-content></ng-content>
  <i *ngIf="icon" class="{{ icon }}"></i>
</button>
