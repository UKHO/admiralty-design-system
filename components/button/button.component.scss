@use "../../styles/base/colours";
@use "../../styles/base/responsive";
@use "../../styles/base/typography";
@use "../../styles/base/icons";

/* Button Wrapper */
.button-wrapper {
  display: inline-flex;
  flex-direction: column;

  .error-wrap {
    max-width: 240px;
    display: flex;
    margin-top: 11px;

    > span {
      color: colours.$colour-utility-error;
      margin-left: 10px;
      font-size: 16px;
    }
  }
}

@mixin button-defaults {
  display: inline-flex;
  align-items: center;
  justify-content: center;

  font-size: 21px;
  font-weight: typography.$font-weight-medium;
  background-color: unset;
  cursor: pointer;

  &:focus {
    outline: 3px solid colours.$colour-utility-focus;
  }

  &:disabled {
    cursor: default;
    * {
      pointer-events: none;
    }
  }

  i {
    display: inline-block;
    font-size: 24px;
    color: colours.$colour-button-icon;
    padding-left: 15px;
  }
}

@mixin button {
  @include button-defaults;
  height: 60px;
  padding: 15px 18px;
  white-space: nowrap;
  min-width: 200px;

  @include responsive.not-desktop {
    min-width: unset;
    max-width: 200px;
    width: 100%;
    height: 50px;
    padding: 10px 18px;
  }

  i {
    display: inline-block;
    font-size: 24px;
    color: colours.$colour-button-icon;
    padding-left: 15px;
  }
}

/* Primary Buttons */
ukho-button button.primary,
ukho-button input[type="button"].primary {
  @include button;
  background-color: colours.$colour-button-standard;
  color: colours.$colour-button-text;
  border: 2px solid colours.$colour-button-standard;

  &:hover {
    background-color: colours.$colour-button-hover;
    border: 2px solid colours.$colour-button-hover;
  }

  &:active {
    background-color: colours.$colour-button-selected;
    border: 2px solid colours.$colour-button-border-selected;
    color: colours.$colour-button-icon-selected;

    i {
      color: colours.$colour-button-icon-selected;
    }

    p {
      color: colours.$colour-button-text-selected;
    }
  }

  &:disabled {
    background-color: colours.$colour-button-deselected;
    border: 2px solid colours.$colour-button-deselected;
  }
}

/* Secondary Buttons */
ukho-button button.secondary,
ukho-button input[type="button"].secondary {
  @include button;
  background-color: colours.$colour-button-secondary-standard;
  color: colours.$colour-button-secondary-text;
  border: 2px solid colours.$colour-button-secondary-border-standard;

  &:hover {
    background-color: colours.$colour-button-secondary-hover;
    border: 2px solid colours.$colour-button-secondary-border-hover;
  }

  &:active {
    background-color: colours.$colour-button-secondary-selected;
    border: 2px solid colours.$colour-button-secondary-selected;
    color: colours.$colour-button-secondary-icon-selected;

    i {
      color: colours.$colour-button-secondary-icon-selected;
    }

    p {
      color: colours.$colour-button-secondary-text-selected;
    }
  }

  &:disabled {
    background-color: colours.$colour-button-secondary-deselected;
    border: 2px solid colours.$colour-button-secondary-deselected;
  }

  i {
    color: colours.$colour-button-secondary-icon;
  }
}

/* Warning Buttons */
ukho-button button.warning,
ukho-button input[type="button"].warning {
  @include button;
  background-color: colours.$colour-utility-error;
  color: colours.$colour-white;
  border: 2px solid colours.$colour-utility-error;

  &:hover {
    background-color: colours.darken(colours.$colour-utility-error);
    border: 2px solid colours.darken(colours.$colour-utility-error);
  }

  &:active {
    background-color: colours.$colour-white;
    border: 2px solid colours.$colour-utility-error;
    color: colours.$colour-utility-error;

    i {
      color: colours.$colour-utility-error;
    }

    p {
      color: colours.$colour-utility-error;
    }
  }

  &:disabled {
    background-color: colours.$colour-button-secondary-deselected;
    border: 2px solid colours.$colour-button-secondary-deselected;
    & > * {
      opacity: 0.5;
    }
  }

  i {
    color: colours.$colour-white;
  }
}

/* Text Buttons */
ukho-button button.text,
ukho-button input[type="button"].text {
  @include button-defaults;
  color: colours.$colour-button-secondary-text;

  &:active {
    color: colours.$colour-button-secondary-text;

    i {
      color: colours.$colour-button-secondary-text;
    }

    p {
      color: colours.$colour-button-secondary-text;
    }
  }

  &:disabled {
    color: colours.$colour-button-secondary-deselected;
    & > * {
      opacity: 0.5;
    }
  }

  i {
    color: colours.$colour-button-secondary-icon;
  }
}

/* Icon Buttons */
ukho-button button.icon,
ukho-button input[type="button"].icon {
  @include button-defaults;
  min-height: 50px;
  min-width: 50px;
  padding: 10px 10px;

  background-color: colours.$colour-bg-grey;
  color: colours.$colour-admiralty-blue;
  border: 2px solid colours.$colour-admiralty-blue;

  &:active {
    background-color: colours.$colour-admiralty-blue;
    color: colours.$colour-white;
  }

  &:disabled {
    opacity: 0.3;
  }

  &:active:disabled {
    background-color: colours.$colour-bg-grey;
  }

  i {
    margin: auto;
    font-size: unset;
    color: unset;
    padding-left: unset;
  }
}
