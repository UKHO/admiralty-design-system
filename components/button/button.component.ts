import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'ukho-button',
  templateUrl: './button.component.html',
  styleUrls: ['./button.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonComponent {
  @Input() secondary = false;
  @Input() icon: string;
  @Input() disabled = false;
}
