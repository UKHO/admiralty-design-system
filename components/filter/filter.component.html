<section class="filters">
  <header class="heading">
    <h2>{{ heading }}</h2>
    <div class="button-wrapper">
      <button type="reset" class="text-link" (click)="onClear()">Clear</button>
    </div>
  </header>
  <ukho-expansion
    *ngFor="let group of groups; let groupIndex = index; trackBy: trackExpansion"
    (change)="onExpansionChange(group)"
    [hideBorder]="true"
    [heading]="group.title"
    [initialExpanded]="group.expanded"
  >
    <nav class="filter-items" [attr.aria-label]="group.title">
      <ukho-checkbox
        *ngFor="let item of group.items; trackBy: trackFilterItem.bind(this, groupIndex)"
        (change)="onCheckboxChange($event, group, item)"
        [checkboxRight]="true"
        [checked]="item.selected"
        [justify]="true"
        >{{ item.title }}</ukho-checkbox
      >
    </nav>
  </ukho-expansion>
  <ukho-button (click)="onApply()">Apply filters</ukho-button>
</section>
