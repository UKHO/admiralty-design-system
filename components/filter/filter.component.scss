.filters {
  display: table;

  .heading {
    display: flex;
    justify-content: space-between;
    align-items: baseline;

    h2 {
      margin-bottom: 0;
    }

    /* TODO: If reused, move class to button component */
    button.text-link {
      background-color: unset;
      border: none;
      color: black;
      margin-left: 28px;
      padding: 0;
      min-width: unset;
      width: unset;
      height: unset;
      font-size: 16px;
      text-decoration: underline;
      font-weight: unset;
    }
    button.text-link:hover {
      background-color: unset;
      border: unset;
    }
  }

  ukho-expansion {
    display: block;
    margin-left: 7px;
  }

  ukho-checkbox {
    display: flex;
  }

  .filter-items {
    display: flex;
    flex-direction: column;
    margin-left: 15px;
  }

  ukho-button {
    display: block;
    margin-top: 10px;
  }
}
