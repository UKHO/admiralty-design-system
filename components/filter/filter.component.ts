import { Component, Input, Output, EventEmitter } from '@angular/core';
import { FilterGroup, FilterItem } from './filter.types';

@Component({
  selector: 'ukho-filter',
  templateUrl: './filter.component.html',
  styleUrls: ['./filter.component.scss'],
})
export class FilterComponent {
  @Input() groups: FilterGroup[];
  @Input() heading: string = 'Filters';

  /**
   * An event that is emitted when the Apply Filters button is pressed.
   *
   * The event will contain an array of filter groups.
   */
  @Output() filtersChange = new EventEmitter<FilterGroup[]>();

  // Return a unique ID for each expansion so Angular can track changes
  trackExpansion = (index: number) => `expansion-${index}`;
  // Return a unique ID for each filter item so Angular can track changes
  trackFilterItem = (groupIndex: number, index: number) => `expansion-${groupIndex}-item-${index}`;

  onExpansionChange(filterGroup: FilterGroup) {
    this.groups = this.groups.map((group: FilterGroup) => {
      if (group.title !== filterGroup.title) {
        return group;
      }
      const { expanded, ...rest } = group;
      return { ...rest, ...(!filterGroup.expanded && { expanded: true }) };
    });
  }

  onCheckboxChange(event: Event, filterGroup: FilterGroup, filterItem: FilterItem) {
    event.stopPropagation();
    this.groups = this.groups.map((group: FilterGroup) => {
      if (group.title !== filterGroup.title) {
        return group;
      }
      const items = group.items.map((item: FilterItem) => {
        if (item.title !== filterItem.title) {
          return item;
        }
        const { selected, ...rest } = item;
        return { ...rest, ...(!filterItem.selected && { selected: true }) };
      });
      return { ...group, items };
    });
  }

  onClear(): void {
    this.groups = this.groups.map((group: FilterGroup) => {
      const items = group.items.map((item: FilterItem) => {
        const { selected, ...rest } = item;
        return rest;
      });
      return { ...group, items };
    });
  }

  onApply(): void {
    this.filtersChange.emit(this.groups);
  }
}
