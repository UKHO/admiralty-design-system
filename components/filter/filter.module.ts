import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ButtonModule } from '../button/button.module';
import { CheckboxModule } from '../checkbox/checkbox.module';
import { ExpansionModule } from '../expansion/expansion.module';
import { FilterComponent } from './filter.component';

export { FilterComponent } from './filter.component';

@NgModule({
  declarations: [FilterComponent],
  exports: [FilterComponent],
  imports: [CommonModule, ButtonModule, CheckboxModule, ExpansionModule],
})
export class FilterModule {}
