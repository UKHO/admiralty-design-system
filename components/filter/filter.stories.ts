import { FilterComponent } from './filter.component';
import { FilterGroup } from './filter.types';
import { action } from '@storybook/addon-actions';
import { Story } from '@storybook/angular';
import { ButtonModule } from '../button/button.module';
import { CheckboxModule } from '../checkbox/checkbox.module';
import { ExpansionModule } from '../expansion/expansion.module';

export default {
  title: 'Filter',
  component: FilterComponent,
  argTypes: {
    groups: {
      table: {
        disable: true,
      },
    },
  },
};

const mockGroups: FilterGroup[] = [
  {
    title: 'Product',
    items: [
      {
        title: 'AVCS',
      },
      {
        title: 'ADP',
      },
      {
        title: 'eNP',
      },
    ],
  },
  {
    title: 'Media type',
    items: [
      {
        title: 'DVD',
      },
      {
        title: 'CD',
      },
      {
        title: 'BASE',
      },
      {
        title: 'ZIP',
      },
    ],
  },
  {
    title: 'Week number',
    items: [
      {
        title: 'Week 39',
      },
      {
        title: 'Week 40',
      },
    ],
  },
  {
    title: 'Year',
    items: [
      {
        title: '2022',
      },
      {
        title: '2021',
      },
      {
        title: '2020',
      },
    ],
  },
];

const Template: Story<FilterComponent> = (args: FilterComponent) => ({
  props: { ...args, filtersChange: action('Filters changed') },
  moduleMetadata: {
    declarations: [FilterComponent],
    imports: [ButtonModule, CheckboxModule, ExpansionModule],
  },
  template: `<ukho-filter [groups]="groups" [heading]="heading" (filtersChange)="filtersChange($event)"></ukho-filter>`,
});

export const Empty: Story = Template.bind({});

export const Default: Story = Template.bind({});
Default.args = {
  groups: mockGroups,
};

export const Selected: Story = Template.bind({});
Selected.args = {
  groups: [
    {
      ...mockGroups[0],
      expanded: true,
      items: [
        { ...mockGroups[0].items[0], selected: true },
        { ...mockGroups[0].items[1], selected: true },
        ...mockGroups[0].items.slice(2),
      ],
    },
    {
      ...mockGroups[1],
      expanded: true,
      items: [
        ...mockGroups[1].items.slice(0, 1),
        { ...mockGroups[1].items[1], selected: true },
        { ...mockGroups[1].items[2], selected: true },
        ...mockGroups[1].items.slice(3),
      ],
    },
    ...mockGroups.slice(2),
  ],
};

export const Constrained: Story<FilterComponent> = () => ({
  props: {
    groups: [...mockGroups.slice(0, 3), { ...mockGroups[3], title: 'Year of publication' }],
    filtersChange: action('Filters changed'),
  },
  moduleMetadata: {
    declarations: [FilterComponent],
    imports: [ButtonModule, CheckboxModule, ExpansionModule],
  },
  template: `<div style="width:160px;"><ukho-filter [groups]="groups" (filtersChange)="filtersChange($event)"></ukho-filter></div>`,
});

export const AlternateHeading: Story = Template.bind({});
AlternateHeading.args = {
  groups: mockGroups,
  heading: 'Refine',
};