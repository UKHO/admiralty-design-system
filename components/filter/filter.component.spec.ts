import { By } from '@angular/platform-browser';
import { createHostFactory } from '@ngneat/spectator';
import { MockComponent } from 'ng-mocks';

import { FilterComponent } from './filter.component';
import { FilterGroup } from './filter.types';

import { ButtonComponent } from '../button/button.module';
import { CheckboxComponent } from '../checkbox/checkbox.component';
import { ExpansionComponent } from '../expansion/expansion.module';

describe('FilterComponent', () => {
  const createHost = createHostFactory({
    component: FilterComponent,
    declarations: [
      FilterComponent,
      MockComponent(ButtonComponent),
      MockComponent(CheckboxComponent),
      MockComponent(ExpansionComponent),
    ],
  });

  const mockGroups: FilterGroup[] = [
    {
      title: 'Product',
      items: [
        {
          title: 'AVCS',
        },
        {
          title: 'ADP',
        },
      ],
    },
    {
      title: 'Media type',
      items: [
        {
          title: 'DVD',
        },
        {
          title: 'CD',
        },
        {
          title: 'BASE',
        },
        {
          title: 'ZIP',
        },
      ],
    },
    {
      title: 'Week number',
      items: [
        {
          title: 'Week 39',
        },
        {
          title: 'Week 40',
        },
      ],
    },
    {
      title: 'Year',
      items: [
        {
          title: '2022',
        },
        {
          title: '2021',
        },
        {
          title: '2020',
        },
      ],
    },
  ];

  it('should create', () => {
    const spectator = createHost('<ukho-filter></ukho-filter>');
    expect(spectator.component).toBeTruthy();
  });

  it('should render the groups', () => {
    const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
      hostProps: {
        groups: mockGroups,
      },
    });

    const expansionElements = spectator.queryAll('ukho-expansion');
    expect(expansionElements.length).toBe(4);
    const itemElements = spectator.queryAll('ukho-checkbox');
    expect(itemElements.length).toBe(11);
  });

  it('should render the groups with inital expansion and selection values set', () => {
    const groups: FilterGroup[] = [
      {
        ...mockGroups[0],
        items: [{ ...mockGroups[0].items[0], selected: true }, ...mockGroups[0].items.slice(1)],
        expanded: true,
      },
      ...mockGroups.slice(1),
    ];
    const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
      hostProps: {
        groups,
      },
    });

    const expansionEl = spectator.fixture.debugElement.queryAll(By.directive(ExpansionComponent));
    expect(expansionEl).toBeTruthy();

    const expansionEl1: ExpansionComponent = expansionEl[0].componentInstance;
    expect(expansionEl1.initialExpanded).toBeTruthy;

    const expansionEl2: ExpansionComponent = expansionEl[1].componentInstance;
    expect(expansionEl2.initialExpanded).toBeFalsy;

    const checkboxEl = spectator.fixture.debugElement.queryAll(By.directive(CheckboxComponent));
    expect(checkboxEl).toBeTruthy();

    const checkbox1: CheckboxComponent = checkboxEl[0].componentInstance;
    expect(checkbox1.checked).toBeTruthy;

    const checkbox2: CheckboxComponent = checkboxEl[1].componentInstance;
    expect(checkbox2.checked).toBeFalsy;
  });

  describe('onExpansionChange', () => {
    it('should expand the first group', () => {
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups: mockGroups,
        },
      });

      spectator.component.onExpansionChange(mockGroups[0]);

      expect(spectator.component.groups).toEqual([{ ...mockGroups[0], expanded: true }, ...mockGroups.slice(1)]);
    });

    it('should expand the second group', () => {
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups: mockGroups,
        },
      });

      spectator.component.onExpansionChange(mockGroups[1]);

      expect(spectator.component.groups).toEqual([
        ...mockGroups.slice(0, 1),
        { ...mockGroups[1], expanded: true },
        ...mockGroups.slice(2),
      ]);
    });

    it('should collapse the first group', () => {
      const groups: FilterGroup[] = [{ ...mockGroups[0], expanded: true }, ...mockGroups.slice(1)];
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups,
        },
      });

      spectator.component.onExpansionChange(groups[0]);

      expect(spectator.component.groups).toEqual(mockGroups);
    });
  });

  describe('onCheckboxChange', () => {
    it('should select the first item in the first group', () => {
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups: mockGroups,
        },
      });

      spectator.component.onCheckboxChange(mockGroups[0], mockGroups[0].items[0]);

      expect(spectator.component.groups).toEqual([
        { ...mockGroups[0], items: [{ ...mockGroups[0].items[0], selected: true }, ...mockGroups[0].items.slice(1)] },
        ...mockGroups.slice(1),
      ]);
    });

    it('should unselect the first item in the first group', () => {
      const groups: FilterGroup[] = [
        { ...mockGroups[0], items: [{ ...mockGroups[0].items[0], selected: true }, ...mockGroups[0].items.slice(1)] },
        ...mockGroups.slice(1),
      ];
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups,
        },
      });

      spectator.component.onCheckboxChange(groups[0], groups[0].items[0]);

      expect(spectator.component.groups).toEqual(mockGroups);
    });
  });

  describe('onClear', () => {
    it('should clear all selections but retain expansions', () => {
      const groups: FilterGroup[] = [
        {
          ...mockGroups[0],
          expanded: true,
          items: [
            { ...mockGroups[0].items[0], selected: true },
            { ...mockGroups[0].items[1], selected: true },
            ...mockGroups[0].items.slice(2),
          ],
        },
        {
          ...mockGroups[1],
          items: [
            ...mockGroups[1].items.slice(0, 1),
            { ...mockGroups[1].items[1], selected: true },
            ...mockGroups[1].items.slice(2),
          ],
        },
        ...mockGroups.slice(2),
      ];
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups,
        },
      });

      expect(spectator.component.groups).toEqual(groups);

      spectator.component.onClear();

      expect(spectator.component.groups).toEqual([{ ...mockGroups[0], expanded: true }, ...mockGroups.slice(1)]);
    });
  });

  describe('onClear', () => {
    it('should emit an event with the current filters', () => {
      const groups: FilterGroup[] = [
        {
          ...mockGroups[0],
          expanded: true,
          items: [
            { ...mockGroups[0].items[0], selected: true },
            { ...mockGroups[0].items[1], selected: true },
            ...mockGroups[0].items.slice(2),
          ],
        },
        {
          ...mockGroups[1],
          items: [
            ...mockGroups[1].items.slice(0, 1),
            { ...mockGroups[1].items[1], selected: true },
            ...mockGroups[1].items.slice(2),
          ],
        },
        ...mockGroups.slice(2),
      ];
      const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
        hostProps: {
          groups,
        },
      });
      const filtersChangeSpy = jest.spyOn(spectator.component.filtersChange, 'emit');

      expect(spectator.component.groups).toEqual(groups);
      expect(filtersChangeSpy).not.toBeCalled();

      spectator.component.onApply();
      expect(filtersChangeSpy).toBeCalledTimes(1);
      expect(filtersChangeSpy).toBeCalledWith(groups);
    });
  });

  it('should create compenent with expected structure', () => {
    const spectator = createHost('<ukho-filter [groups]="groups"></ukho-filter>', {
      hostProps: {
        groups: mockGroups,
      },
    });

    expect(spectator.fixture).toMatchSnapshot();
  });
});
