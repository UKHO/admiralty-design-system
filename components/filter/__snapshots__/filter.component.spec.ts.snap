// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComponent should create compenent with expected structure 1`] = `
<ng-component
  groups={[Function Array]}
>
  <ukho-filter>
    <section
      class="filters"
    >
      <header
        class="heading"
      >
        <h2>
          Filters
        </h2>
        <div
          class="button-wrapper"
        >
          <button
            class="text-link"
            type="reset"
          >
            Clear
          </button>
        </div>
      </header>
      
      <ukho-expansion>
        <nav
          aria-label="Product"
          class="filter-items"
        >
          
          <ukho-checkbox>
            AVCS
          </ukho-checkbox>
          <ukho-checkbox>
            ADP
          </ukho-checkbox>
        </nav>
      </ukho-expansion>
      <ukho-expansion>
        <nav
          aria-label="Media type"
          class="filter-items"
        >
          
          <ukho-checkbox>
            DVD
          </ukho-checkbox>
          <ukho-checkbox>
            CD
          </ukho-checkbox>
          <ukho-checkbox>
            BASE
          </ukho-checkbox>
          <ukho-checkbox>
            ZIP
          </ukho-checkbox>
        </nav>
      </ukho-expansion>
      <ukho-expansion>
        <nav
          aria-label="Week number"
          class="filter-items"
        >
          
          <ukho-checkbox>
            Week 39
          </ukho-checkbox>
          <ukho-checkbox>
            Week 40
          </ukho-checkbox>
        </nav>
      </ukho-expansion>
      <ukho-expansion>
        <nav
          aria-label="Year"
          class="filter-items"
        >
          
          <ukho-checkbox>
            2022
          </ukho-checkbox>
          <ukho-checkbox>
            2021
          </ukho-checkbox>
          <ukho-checkbox>
            2020
          </ukho-checkbox>
        </nav>
      </ukho-expansion>
      <ukho-button>
        Apply filters
      </ukho-button>
    </section>
  </ukho-filter>
</ng-component>
`;
